components/nexus-mime/src/test/java/org/sonatype/nexus/mime/internal/DefaultMimeSupportTest.java

```diff
@@ -87,35 +75,33 @@ public void testGuessfakeMimeRulesSourceMimeTypeFromPath() {
     MimeRulesSource source = new MimeRulesSource()
     {
       @Override
-      public String getRuleForPath(String path) {
-        return "foo/bar";
+      public MimeRule getRuleForName(String path) {
+        return new MimeRule(false, "foo/bar");
       }
     };
 
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/artifact.pom"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/artifact.jar"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/artifact-sources.jar"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/maven-metadata.xml"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.xml"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.tar.gz"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.tar.bz2"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.zip"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.war"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.ear"), equalTo("foo/bar"));
-    assertThat(underTest.guessMimeTypeFromPath(source, "/some/path/some.ejb"), equalTo("foo/bar"));
+    assertThat(underTest.guessMimeTypeFromPath("/some/path/artifact.pom", source), equalTo("foo/bar"));
```

Comment:
not sure I get the point of this exhaustive testing since in all cases the path is completely ignored...